kendzi3d-jogl-model-render/src/main/java/kendzi/jogl/model/render/ModelRender.java

```diff
@@ -142,53 +160,54 @@ private void draw(GL2 gl, Model model) {
                     gl.glEnd();
                 }
 
-                unsetupTextures(gl, material, useTextures);
-
+                if (drawTextures) {
+                    if (model.useTextureAlpha)
+                        disableTransparentText(gl);
+                    unsetupTextures(gl, material, mesh.hasTexture);
+                }
             }
 
             gl.glColor3f(1.0f, 1.0f, 1.0f);
 
         } catch (RuntimeException e) {
             throw new RuntimeException("error model: " + model.getSource() + " mesh: " + mi + " ("
                     + (model.mesh[mi] != null ? model.mesh[mi].name : "") + ")" + " face: " + fi, e);
-        }
+        } finally {
 
-        gl.glDisable(GLLightingFunc.GL_LIGHTING);
+            gl.glLightModeli(GL2.GL_LIGHT_MODEL_TWO_SIDE, GL2.GL_FALSE);
 
```

Comment:
Here you actually catch and re-throw the RuntimeException, but I have not seen it yet land on the console.

I have done quite some testing, so the two possibilities are 1) the code is ok and it does not occur  or  2) the newly thrown exception does not make it to the console for us to be noticed.    I suppose this is a good point to generate a test case for, i.e. produce a faulty model that raises such an exception on purpose and check if it lands in System.out or System.err (or a dispatching logging facility).